src/components/PopupWithForm.js

```diff
@@ -8,26 +8,44 @@ export default class PopupWithForm extends Popup {
     this._inputs = Array.from(
       this._popupForm.querySelectorAll(".modal__input")
     );
+    this._submitButton = this._popupForm.querySelector(".modal__button");
+    this._submitButtonText =
+      this._submitButton.childNodes[0].textContent.trim();
   }
 
   _getInputValues() {
     this._inputValues = {};
     this._inputs.forEach((input) => {
       this._inputValues[input.name] = input.value;
     });
-
     return this._inputValues;
   }
 
+  setInputValues(data) {
+    console.log("Setting input values with data:", data);
+    this._inputs.forEach((input) => {
+      input.value = data[input.name] || "";
+    });
+  }
+
   setEventListeners() {
     super.setEventListeners();
-
-    this._submitHandler = (evt) => {
+    this._popupForm.addEventListener("submit", (evt) => {
       evt.preventDefault();
+      const result = this._handleFormSubmit(this._getInputValues());
+      if (result && typeof result.then === "function") {
+        result.then(() => this.close()).catch(console.error);
+      } else {
+        console.error("handleFormSubmit did not return a promise");
```

Comment:
You don't need it because you have everything in `index.js`

You'll have only this

```js


  setEventListeners() {
    super.setEventListeners();
    this._popupForm.addEventListener("submit", (evt) => {
      evt.preventDefault();
      this._handleFormSubmit(this._getInputValues());
    });
  }

```